src/com/palmergames/bukkit/towny/command/NationCommand.java

```diff
@@ -1208,6 +1213,14 @@ public void nationAdd(Player player, String[] names) throws TownyException {
 				removeinvites.add(townname);
 				continue;
 			}
+
+            if (TownySettings.getMaxResidentsPerNation() > 0 && 
+                nation.getResidents().size() + TownyAPI.getInstance().getTown(townname).getResidents().size() >= TownySettings.getMaxResidentsPerNation()) {
+                // Town has too many residents to join the nation
+                removeinvites.add(townname);
+                continue;

```

Comment:
this should TownyMessaging.sendErrorMsg(.....) with a new language string that says X town couldn't be added because then the nation would have too many residents.